pagefaultgames/pokerogue
# [QoL] Added the ability for tandemaus to evolve into a 3 family maushold

## What are the changes?
Currently tandemaus will only evolve into a 4 family maushold. This PR fixes it so that when evolving, there's a 1/4 chance of getting a 3 family maushold and a 3/4 chance to get a 4 family maushold

## Why am I doing these changes?
It was brought up on the discord that this functionality was missing

## What did change?
Change tandemaus' evolution in pokemon-evolution to be like dunsparce in that there's a 1/4 chance of getting one form (3 mouse family) and a 3/4 chance of getting a 4 mouse family.

Also it was found the 3 and 4 family forms of maushold had their weights swapped, likely from an old copy/paste thing; this has been fixed too.

Lastly added a test (thanks to some amazing people on discord who helped with it) to check that the functionality is working properly

### Screenshots/Videos
Tandemaus evolving into 3 mouse family:

https://github.com/pagefaultgames/pokerogue/assets/66582645/778f15ca-91e8-4aa0-9e80-c5e54a42f73f

Tandemaus evolving into 4 mouse family:

https://github.com/pagefaultgames/pokerogue/assets/66582645/8c1b44fb-20ed-4ffe-a0d0-197a99322e86

## How to test the changes?

Update line 35 of the overrides.ts files to force single battles, like so:
`export const SINGLE_BATTLE_OVERRIDE: boolean = true;`

Download the save file from [here](https://discord.com/channels/1125469663833370665/1176874654015684739/1257263993299406899). In it, I am on wave 55 of endless, and I have a full party of tandemaus, except my starting pokemon which is a maushold. These are the steps you need to do:

1) Swap out your maushold with the top tandemaus when prompted when starting the wave. It will be the tandemaus with 60HP, like so:

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/98b75c10-1281-4c0a-8fb9-692e7115a3b1)

2) Use a master ball to catch the enemy tandemaus and replace your maushold with it.

3) You should get a bunch of exp and level up to level 34. You'll be asked if you want to learn "Charm" due to leveling up but you can say no

4) Your tandemaus should now evolve into a 3 family maushold, like so:

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/083053da-a988-4bab-a3da-fc2aa30e52b0)

5) From here on out, at the start of every wave put out a tandemaus and use a master ball on each wave to instantly catch the enemy. Don't take any moves to save time. You should see other tandemaus evolving into 4 family mausholds in the coming waves.

You have 5 tandemaus to start off with, so feel free to do it as many times as you like, but if you'd like to continue, you can change these overrides too:

Line 92 forces the opponent to always be tandemaus: 
`export const OPP_SPECIES_OVERRIDE: Species | integer = Species.TANDEMAUS;`

Line 93 forces the opponent to be level 24 (one level before leveling): 
`export const OPP_LEVEL_OVERRIDE: number = 24;`

If you are using the save file provided, you will have dozens of master balls you can use to keep catching tandemaus and leveling them up.

## Checklist
